src/choose-location/choose-location.component.tsx

```diff
@@ -48,160 +44,47 @@ export default function ChooseLocation(props: ChooseLocationProps) {
           }
         })
         .catch(createErrorHandler());
-    }
-    return () => abortController.abort();
-  }, [isSubmitting]);
+    },
+    [referrer]
```

Comment:
Deps are optional for `useCallback`? I've never used it before and wouldn't know. But if this were `useEffect` you'd need to include `props` as a dep, I think.

Reply:
The deps are only markers for changes. If you disregard changes in certain deps you should not include them in the deps. Here I just want to capture the initial `props`, not any changed ones.

Don't use deps blindly. They are essential, but understanding what they do and *when* to use them is even more essential.